ui/app/src/modules/Preview/Tools/AudiencesPanel.tsx

```diff
@@ -27,12 +86,321 @@ const translations = defineMessages({
   }
 });
 
-export default function AudiencesPanel() {
+interface AudiencesPanelUIProps {
+  config: any;
+  profile: any;
+  onFormChange: Function;
+  saveProfile: Function;
+  setDefaults: Function;
+}
+
+export function AudiencesPanelUI(props: AudiencesPanelUIProps) {
+
+  const classes = useStyles({});
+  const {
+    config,
+    profile,
+    onFormChange,
+    saveProfile,
+    setDefaults
+  } = props;
+
   return (
     <ToolPanel title={translations.audiencesPanel}>
-      <Typography component="h2" variant="subtitle1" style={{ padding: '10px' }}>
-        Audiences Panel
-      </Typography>
+      {
+        config ? (
+          <>
+            <Grid className={classes.PanelMargin}>
+              {
+                config.map((property: any, index: number) => (
+                  <div key={index}>
+                    <GetCodeDependingType
+                      property={property}
+                      profile={profile}
+                      onFormChange={onFormChange}
+                    />
+                    <Divider className={classes.divider} />
+                  </div>
+                ))
+              }
+            </Grid>
+            <Grid className={classes.actionBTN} >
+              <Button variant="contained" onClick={() => setDefaults()}>
+                <FormattedMessage
+                  id="audiencesPanel.defaults"
+                  defaultMessage={`Defaults`}
+                />
+              </Button>
+              <Button variant="contained" color="primary" onClick={() => saveProfile()}>
+                <FormattedMessage
+                  id="audiencesPanel.apply"
+                  defaultMessage={`Apply`}
+                />
+              </Button>
+            </Grid>
+          </>
+        ) : (null)
+      }
     </ToolPanel>
   );
 }
+
+export default function AudiencesPanel() {
+
+  const [config, setConfig] = useState();
+  const [profile, setProfile] = useState();
+  const site = useSelector<GlobalState, GlobalState['sites']>(state => state.sites).active;
+
+  useEffect(() => {
+
+    forkJoin([
+      getAudiencesPanelConfig(site),
```

Comment:
Move the data fetching to PreviewConcierge. Follow ComponentsPanel example. **Ask early if you’re disoriented.**